File: backend/app/schemas/workout.py
"""Workout API의 request/response validation schema."""

from datetime import date, datetime
from uuid import UUID

from pydantic import AwareDatetime, BaseModel, Field, model_validator


class WorkoutSessionCreate(BaseModel):
    workout_date: date
    started_at: AwareDatetime
    ended_at: AwareDatetime
    workout_seconds: float = Field(ge=0)
    squat_count: int = Field(ge=0)
    stretch_seconds: float = Field(ge=0)

    @model_validator(mode="after")
    def validate_time_order(self) -> "WorkoutSessionCreate":
        if self.ended_at < self.started_at:
            raise ValueError("ended_at must be greater than or equal to started_at")
        return self


class WorkoutSessionCreatedResponse(BaseModel):
    id: UUID


class WorkoutSessionResponse(BaseModel):
    id: UUID
    workout_date: date
    started_at: datetime
    ended_at: datetime
    workout_seconds: float
    squat_count: int
    stretch_seconds: float


class DailyWorkoutSummaryResponse(BaseModel):
    workout_date: date
    squat_count: int
    stretch_seconds: float
    workout_seconds: float
    session_count: int

File: frontend/src/types/workout.ts
export interface DailyWorkoutSummary {
  workout_date: string
  squat_count: number
  stretch_seconds: number
  workout_seconds: number
  session_count: number
}

export interface WorkoutSession {
  id: string
  workout_date: string
  started_at: string
  ended_at: string
  workout_seconds: number
  squat_count: number
  stretch_seconds: number
}
